src/gradualizer_task.erl

```diff
@@ -0,0 +1,184 @@
+%% @doc Task is a helper for running asynchronous operations.
+%%
+%% This is a port of the Elixir Task module borrowed from https://github.com/redink/task.
+%% Original Elixir Task documentation can be found at https://hexdocs.pm/elixir/Task.html.
+%% @end
+-module(gradualizer_task).
```

Comment:
If this would be part of the standard library (as it is in Elixir) I'd be fine using it. Porting it and including it here seems like lots of added complexity though. Isn't it enough to use spawn_link or spawn_monitor and set a timeout?

```Erlang
%% Pseudo-code
TypecheckerPid = spawn_link(fun () -> type_check(Froms, self()) end),
receive
    {typechecker_done, Result} -> Result;
after Timeout ->
    kill(TypecheckerPid),
    {timeout, Forms)
end.
```

Reply:
It worked pretty well for me, but we now have a hand rolled `async/3` if you prefer 🤷‍♂️

Reply:
@zuiderkwast you must've got some sixth sense 😆 It turned out the `task.erl` would return with a timeout, so other forms would be checked and reported, but at the same time the async task would go rouge and hog the CPU in an infinite loop anyway. I've now tested in ErlangLS with the hand-rolled `spawn_monitor` and timeout and the CPU is way less busy 👍